File: frontend/src/api/preprocessing.js
import apiClient from './client';

/**
 * Fetches auto-detected datetime and numeric target column recommendations.
 * @param {string} datasetId 
 * @returns {Promise<Object>} ColumnRecommendations
 */
export const getRecommendations = async (datasetId) => {
  const response = await apiClient.get(`/api/preprocessing/recommendations/${datasetId}`);
  return response.data;
};

/**
 * Runs the time-series chronological sorting, imputation, and feature engineering pipeline.
 * @param {Object} payload 
 * @returns {Promise<Object>} PreprocessingResponse
 */
export const runPreprocessing = async (payload) => {
  const response = await apiClient.post('/api/preprocessing/process', payload);
  return response.data;
};

File: backend/app/api/preprocessing.py
from fastapi import APIRouter
from app.schemas.preprocessing import (
    PreprocessingRequest,
    PreprocessingResponse,
    ColumnRecommendations,
)
from app.services.preprocessing_service import (
    get_column_recommendations,
    execute_preprocessing,
)

router = APIRouter(prefix="/preprocessing", tags=["Preprocessing"])


@router.get("/recommendations/{dataset_id}", response_model=ColumnRecommendations)
def get_recommendations(dataset_id: str) -> ColumnRecommendations:
    """
    Analyzes dataset columns and returns auto-detected recommendations for
    datetime column and numeric target column.
    """
    return get_column_recommendations(dataset_id)


@router.post("/process", response_model=PreprocessingResponse)
def run_preprocessing(request: PreprocessingRequest) -> PreprocessingResponse:
    """
    Executes chronological sorting, missing-value handling, and feature engineering
    (calendar features, lag attributes, rolling statistics) on the dataset.
    """
    return execute_preprocessing(request)
